src/util/metadata/global.rs

```diff
@@ -31,171 +30,308 @@ impl MetadataSpec {
             MetadataSpec::InHeader(_) => panic!("Expect a side spec"),
         }
     }
-}
 
-/// A function to load the specified metadata's content.
-///
-/// # Arguments:
-///
-/// * `metadata_spec`: is one of the const `MetadataSpec` instances from the ObjectModel trait, for the target metadata. Whether the metadata is in-header or on-side is a VM-specific choice.
-/// * `object`: is a reference to the target object.
-/// * `mask`: is an optional mask value for the metadata. This value is used in cases like the forwarding pointer metadata, where some of the bits are reused by other metadata such as the forwarding bits.
-/// * `atomic_ordering`: is an optional atomic ordering for the load operation. An input value of `None` means the load operation is not atomic, and an input value of `Some(Ordering::X)` means the atomic load operation will use the `Ordering::X`.
-///
-/// # Returns the metadata value as a word. If the metadata size is less than a word, the effective value is stored in the low-order bits of the word.
-///
-#[inline(always)]
-pub fn load_metadata<VM: VMBinding>(
-    metadata_spec: &MetadataSpec,
-    object: ObjectReference,
-    mask: Option<usize>,
-    atomic_ordering: Option<Ordering>,
-) -> usize {
-    match metadata_spec {
-        MetadataSpec::OnSide(metadata_spec) => {
-            if let Some(order) = atomic_ordering {
-                side_metadata::load_atomic(metadata_spec, object.to_address(), order)
-            } else {
-                unsafe { side_metadata::load(metadata_spec, object.to_address()) }
+    /// A function to non-atomically load the specified metadata's content.
+    ///
+    /// # Arguments:
+    ///
+    /// * `metadata_spec`: is one of the const `MetadataSpec` instances from the ObjectModel trait, for the target metadata. Whether the metadata is in-header or on-side is a VM-specific choice.
+    /// * `object`: is a reference to the target object.
+    /// * `mask`: is an optional mask value for the metadata. This value is used in cases like the forwarding pointer metadata, where some of the bits are reused by other metadata such as the forwarding bits.
+    ///
+    /// # Returns the metadata value.
```

Comment:
The doc is written in the Markdown format, so this line means a title whose text is "Return the metadata value".  A more natural way to do it should be

```markdown
/// # Return value
///
/// Returns the metadata value
```

Or simply remove the title
```markdown
/// Returns the metadata value.
```

It looks like we have been misusing this in many places.  We don't have to fix it now.  We can tidy up the docs in a separate PR.

Reply:
I moved the 'return' section to the end of the description paragraph (before `#Arguments`). I did the change for the related methods in the `metadata` and the `ObjectModel` module.